python/cudf/cudf/testing/csv.py

```diff
@@ -0,0 +1,115 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+
+
+import logging
+import random
+
+import cudf
+from cudf.testing.io import IOBase
+from cudf.testing.utils import _generate_rand_meta
+from cudf.tests import dataset_generator as dg
+
+logging.basicConfig(
+    format="%(asctime)s %(levelname)-8s %(message)s",
+    level=logging.INFO,
+    datefmt="%Y-%m-%d %H:%M:%S",
+)
+
+
+class CSVReader(IOBase):
+    def __init__(
+        self,
+        file_name="temp_csv.csv",
+        dirs=None,
+        max_rows=100_000,
+        max_columns=1000,
+        max_string_length=None,
+    ):
+        super().__init__(
+            file_name=file_name,
+            dirs=dirs,
+            max_rows=max_rows,
+            max_columns=max_columns,
+            max_string_length=max_string_length,
+        )
+
+    def generate_input(self):
+        if self._regression:
+            (
+                dtypes_meta,
+                num_rows,
+                num_cols,
+                seed,
+                file_name,
+            ) = self.get_next_regression_params()
+        else:
+            seed = random.randint(0, 2 ** 32 - 1)
+            random.seed(seed)
+            dtypes_list = list(cudf.utils.dtypes.ALL_TYPES)
+            dtypes_meta, num_rows, num_cols = _generate_rand_meta(
+                self, dtypes_list
+            )
+            file_name = self._file_name
+            self._current_params["dtypes_meta"] = dtypes_meta
+            self._current_params["file_name"] = self._file_name
+            self._current_params["seed"] = seed
+            self._current_params["num_rows"] = num_rows
+            self._current_params["num_columns"] = num_cols
+        logging.info(
+            f"Generating DataFrame with rows: {num_rows} "
+            f"and columns: {num_cols}"
+        )
+        df = dg.rand_dataframe(dtypes_meta, num_rows, seed).to_pandas()
+        df.to_csv(file_name)
```

Comment:
Could we use a buffer here and in the other Reader class implementations instead of a file as well?

Reply:
Should we be having both mechanisms? So that we have coverage for both code-flows while the fuzz tests are run?

Reply:
I figured the fuzzing was for finding edge / corner cases in kernels which should be run the same way regardless of whether the input was a file or a buffer. I figured by keeping everything in memory we can avoid expensive IO and get more iterations in.

Reply:
Makes sense, changed it :+1: